src/main/scala/li/cil/oc/server/fs/FileSystem.scala

```diff
@@ -57,50 +60,27 @@ object FileSystem extends api.detail.FileSystemAPI {
     path
   }
 
-  override def fromClass(clazz: Class[_], domain: String, root: String): api.fs.FileSystem = {
-    val innerPath = ("/assets/" + domain + "/" + (root.trim + "/")).replace("//", "/")
-
-    val codeSource = clazz.getProtectionDomain.getCodeSource.getLocation.getPath
-    val (codeUrl, isArchive) =
-      if (codeSource.contains(".zip!") || codeSource.contains(".jar!"))
-        (codeSource.substring(0, codeSource.lastIndexOf('!')), true)
-      else
-        (codeSource, false)
-
-    val url = Try {
-      new URL(codeUrl)
-    }.recoverWith {
-      case _: MalformedURLException => Try {
-        new URL("file://" + codeUrl)
-      }
-    }
-    val file = url.map(url => new io.File(url.toURI)).recoverWith {
-      case _: URISyntaxException => url.map(url => new io.File(url.getPath))
-    }.getOrElse(new io.File(codeSource))
+  override def fromResource(loc: ResourceLocation): api.fs.FileSystem = {
+    val innerPath = "/assets/" + loc.getNamespace + "/" + (loc.getPath.trim + "/")
 
-    if (isArchive) {
+    val modInfo = FMLLoader.getLoadingModList().getModFileById(loc.getNamespace)
+    val file = modInfo.getFile().getFilePath().toFile()
```

Comment:
Yes, I can load OpenOS both in standalone client/server with mod jar as well as dev (which used to give me the resources path). In fact, I've had to use resource locations explicitly because in standalone the old classpath-based method got broken by the new `modjar` URL scheme (which lack a `CodeSource`). I'd like to eventually use resource loaders instead of relying on disk reads, but it works fine for now.

Reply:
Awesome, thanks for clarifying :)